### ISSUE TITLE
`/lnurlwallet` hanging forever

### ISSUE THREAD

**Issue by fiatjaf** [lnbits/lnbits#103]
@rootzoll, @arcbtc and I were experiencing this same fatal hang in different setups.

Now I was going to try to debug it but couldn't reproduce anymore. It works all the time for me now even though I'm in the same environment I was two days ago when it didn't.

Please give more information!

**Comment by fiatjaf** [comment]
Ok, it only hangs when you try to redeem an internal lnurl-w voucher.

**Comment by rootzoll** [comment]
So you are able to reproduce it now or you need more debug details from my side?

**Comment by fiatjaf** [comment]
I can reproduce again. Sorry for the repeated requests.

**Comment by fiatjaf** [comment]
Should have been fixed here: https://github.com/lnbits/lnbits/commit/211ac0391bb9b518c978aa45c215ed5d4123652c

I changed the functionality from waiting until the invoice is paid to create an account and then show the page to create an account directly then return the page while simultaneously redeeming the lnurl-withdraw. I don't know if I this was very bad.

Maybe once we have basic lnurl-withdraw support inside the wallet this should all be removed in exchange for that instead so I guess it isn't a big problem.

Also things will improve once I finish my invoice listeners PR.

**Comment by arcbtc** [comment]
Works for me!
How about on the blitz @rootzoll?

**Comment by rootzoll** [comment]
I installed from master and first got a module missing errror for "httpx" and then this one:
```
Sep 30 21:43:01 raspberrypi sh[25789]: Loading .env environment variables…
Sep 30 21:43:03 raspberrypi sh[25789]: Traceback (most recent call last):
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/hypercorn/utils.py", line 92, in load_application
Sep 30 21:43:03 raspberrypi sh[25789]:     module = import_module(import_name)
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/importlib/__init__.py", line 127, in import_module
Sep 30 21:43:03 raspberrypi sh[25789]:     return _bootstrap._gcd_import(name[level:], package, level)
Sep 30 21:43:03 raspberrypi sh[25789]:   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
Sep 30 21:43:03 raspberrypi sh[25789]:   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
Sep 30 21:43:03 raspberrypi sh[25789]:   File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
Sep 30 21:43:03 raspberrypi sh[25789]: ModuleNotFoundError: No module named 'lnbits.app'
Sep 30 21:43:03 raspberrypi sh[25789]: During handling of the above exception, another exception occurred:
Sep 30 21:43:03 raspberrypi sh[25789]: Traceback (most recent call last):
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/bin/hypercorn", line 8, in <module>
Sep 30 21:43:03 raspberrypi sh[25789]:     sys.exit(main())
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/hypercorn/__main__.py", line 265, in main
Sep 30 21:43:03 raspberrypi sh[25789]:     run(config)
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/hypercorn/run.py", line 34, in run
Sep 30 21:43:03 raspberrypi sh[25789]:     worker_func(config)
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/hypercorn/asyncio/run.py", line 178, in asyncio_worker
Sep 30 21:43:03 raspberrypi sh[25789]:     app = load_application(config.application_path)
Sep 30 21:43:03 raspberrypi sh[25789]:   File "/home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/hypercorn/utils.py", line 95, in load_application
Sep 30 21:43:03 raspberrypi sh[25789]:     raise NoAppException()
Sep 30 21:43:03 raspberrypi sh[25789]: hypercorn.utils.NoAppException
Sep 30 21:43:04 raspberrypi systemd[1]: lnbits.service: Main process exited, code=exited, status=1/FAILURE
Sep 30 21:43:04 raspberrypi systemd[1]: lnbits.service: Failed with result 'exit-code'.
```
So something with hypercorn does not look right ... not sure what is causing the hypercorn.utils.NoAppException

**Comment by rootzoll** [comment]
I removed the "QUART_APP" line from the `.env` - see:
https://github.com/rootzoll/raspiblitz/blob/79476da8f210a65324521ced449cf2ac3b2fad1a/home.admin/config.scripts/bonus.lnbits.sh#L262

and now its starting .... BUT website looks broken:
![Bildschirmfoto 2020-09-30 um 22 58 15](https://user-images.githubusercontent.com/2472225/94738964-7c34fd00-0370-11eb-814d-c9d6bbe9e3cc.png)


and got this error on debug-logs:
```
Sep 30 21:53:59 raspberrypi sh[22124]: Loading .env environment variables…
Sep 30 21:54:01 raspberrypi sh[22124]: /home/lnbits/.local/share/virtualenvs/lnbits-xqYMrTZ5/lib/python3.7/site-packages/scss/selector.py:54: FutureWarning: Possible nested set at position 329
Sep 30 21:54:01 raspberrypi sh[22124]:   ''', re.VERBOSE | re.MULTILINE)
Sep 30 21:54:04 raspberrypi sh[22124]: Running on 0.0.0.0:5000 over http (CTRL + C to quit)
```

**Comment by fiatjaf** [comment]
Sorry, I messed up the requirements.

**Comment by rootzoll** [comment]
Any updates coming? What should I change to get it working?

**Comment by fiatjaf** [comment]
I think you didn't run the migrate and assets command after the latest update. Could it have been that?

I've updated the commands sequence on `docs/guides/installation.md` and ran them on fresh lnbits installations and everything seemed fine here.

**Comment by rootzoll** [comment]
@fiatjaf I updated to your latest install instructions and still looks the same broken as above. The migrate and assests is running on the latest master code and before starting hypercorn systemd service.

See here the lastest install script:
https://github.com/rootzoll/raspiblitz/blob/b0167471366368211397f83fdb023632c413f8d9/home.admin/config.scripts/bonus.lnbits.sh#L282

**Comment by fiatjaf** [comment]
The issue there was that `QUART_APP` wasn't set so the `migrate` and `assets` command didn't run and errored instead.

Here is the fix and some other improvements (deleting unnecessary dependencies, removing pipenv) https://github.com/rootzoll/raspiblitz/pull/1622/files.

`LndWalletRest`, the backend used by RaspiBlitz currently, is in a bad shape on current `master`, though. https://github.com/lnbits/lnbits/pull/101 improves it.

**Comment by fiatjaf** [comment]
https://github.com/rootzoll/raspiblitz/pull/1622

**Comment by rootzoll** [comment]
Thanks for the PR ... I am on the latest master with #101 merged and still get:
```
Oct 05 18:16:02 raspberrypi sh[25099]: [2020-10-05 18:16:02 +0100] [25101] [ERROR] Error in ASGI Framework
Oct 05 18:16:02 raspberrypi sh[25099]: Traceback (most recent call last):
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/hypercorn/asyncio/context.py", line 28, in _handle
Oct 05 18:16:02 raspberrypi sh[25099]:     await invoke_asgi(app, scope, receive, send)
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/hypercorn/utils.py", line 219, in invoke_asgi
Oct 05 18:16:02 raspberrypi sh[25099]:     await app(scope, receive, send)
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/quart/app.py", line 2068, in __call__
Oct 05 18:16:02 raspberrypi sh[25099]:     await self.asgi_app(scope, receive, send)
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/quart/app.py", line 2091, in asgi_app
Oct 05 18:16:02 raspberrypi sh[25099]:     await asgi_handler(receive, send)
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/quart_trio/asgi.py", line 16, in __call__
Oct 05 18:16:02 raspberrypi sh[25099]:     async with trio.open_nursery() as nursery:
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/trio/_core/_run.py", line 796, in __aenter__
Oct 05 18:16:02 raspberrypi sh[25099]:     self._scope.__enter__()
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/trio/_core/_ki.py", line 159, in wrapper
Oct 05 18:16:02 raspberrypi sh[25099]:     return fn(*args, **kwargs)
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/trio/_core/_run.py", line 449, in __enter__
Oct 05 18:16:02 raspberrypi sh[25099]:     task = _core.current_task()
Oct 05 18:16:02 raspberrypi sh[25099]:   File "/home/lnbits/lnbits/venv/lib/python3.7/site-packages/trio/_core/_run.py", line 2267, in current_task
Oct 05 18:16:02 raspberrypi sh[25099]:     raise RuntimeError("must be called from async context") from None
Oct 05 18:16:02 raspberrypi sh[25099]: RuntimeError: must be called from async context
```

**Comment by fiatjaf** [comment]
You merged too quick, the new merged PR here on lnbits invalidated that raspiblitz PR. I was going there to change exactly now.

**Comment by fiatjaf** [comment]
Here it is: https://github.com/rootzoll/raspiblitz/pull/1624/files

**Comment by rootzoll** [comment]
OK cool ... now its loading again and it does not look broken anymore :) It still has the button css thing, but thats makeup.

![Bildschirmfoto 2020-10-05 um 19 46 33](https://user-images.githubusercontent.com/2472225/95113983-86297800-0743-11eb-89cf-45df8b74669d.png)

I tested the voucher fallback call and the "hanging forever" is gone. After I press "Claim Bitcoin" I get to a LNbits wallet on my mobile phone that is a "balance of zero". When I enter the wallet I see shortly "processing transaction" but after that it stays at balance zero. Tested another voucher before with Wallet of Satoshi - that one worked.

![Screenshot_20201005-194800_Samsung Internet](https://user-images.githubusercontent.com/2472225/95114361-19fb4400-0744-11eb-983c-6783660891fd.jpg)

How can I help to debug this further?

**Comment by fiatjaf** [comment]
Let's keep this in #99.

**Comment by rootzoll** [comment]
Beside the button padding ... there is still the 20 sats balance from a 2 sats voucher" bug. Should I open a new issue for that?

**Comment by fiatjaf** [comment]
Sorry, hadn't read that. Let me see.

**Comment by rootzoll** [comment]
Thanks :) That fixed the Problem
